### DISCUSSION TITLE
Refresh datatable form a https://github.com/wire-elements/modal

### DISCUSSION THREAD

**Discussion by jordivela007** [rappasoft/laravel-livewire-tables#1067] [Q&A] [answered]
When I delete an element from a (https://github.com/wire-elements/modal), the data table is not updating. WHAT AM I WRONG?

public function delete()

{
        $this->authorize('delete', $this->day);
        Day::where('id', $this->user_id)->delete();
        $this->dispatchBrowserEvent('refreshDatatable'); // NOT WORK
        $this->emit('refreshDatatable'); // NOT WORK
        $this->reset(); // restore the default values of the variables after submit the form
        $this->closeModal();
        $this->alert('success', __('Day'), [
            'position' => 'top-end',
            'timer' => 1000,
            'toast' => true,
            'timerProgressBar' => true,
            'text' =>  __('Deleted Successfully.'),
        ]);
}`

**Comment by lrljoe** [comment, marked as answer]
So your Table Component should preferably have something like the below, sot that you can distinguish which Table Component you're issuing a refresh to:
```php
    /**
     * @var array<mixed>
     */
    protected $listeners = ['refreshDayTable' => '$refresh'];
```

Then in your Modal component, 
```php
        $this->emit("refreshDayTable");
```

If this isn't working, then take a look using DebugBar and see if the emit event is actually firing.  Definitely works emiting from that Modal component to the Data Table though, as I use a modified version of WireElements Modal.
